src/java/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -73,19 +73,49 @@
+        usageShort = ExtractIlluminaBarcodes.USAGE_SUMMARY,
         programGroup = Illumina.class
 )
 public class ExtractIlluminaBarcodes extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Tool to determine the barcode for each read in an Illumina lane.  ";
+    static final String USAGE_DETAILS = "<p> This tool will identify reads containing barcodes and output statistics indicating the numbers " +
+            "of reads containing barcode-matching sequences.</p> " +
+            "<p>Illumina sequences can contain at least two types of barcodes, sample and molecular (index).  Sample barcodes (B in the read structure) are used to " +
+            "demultiplex pooled samples while index barcodes (M in the read structure) are used to differentiate multiple reads of a template when carrying out paired-end sequencing.  Note that this tool only extracts sample (B) and not molecular barcodes (M).</p>" +
+            "<p>Barcodes can be provided in the form of a list (BARCODE_FILE) or a string representing the barcode (BARCODE).  " +
+            "The BARCODE_FILE contains multiple fields including 'barcode_sequence_1', 'barcode_sequence_2' (optional), " +
+            "'barcode_name', and 'library_name'.  In contrast, the BARCODE argument is used for runs with reads containing a single barcode (nonmultiplexed) and can be added directly " +
+            "as a string of text e.g. BARCODE=CAATAGCG.</p>" +
+            "" +
+            "<p>Data is output per lane/tile within the BaseCalls directory with the file name format of 's_{lane}_{tile}_barcode.txt'.  " +
+            "These files contain the following tab-separated columns:" +
+            "<ul> " +
+            "<li>Read subsequence at barcode position</li>" +
+            "<li>Y or N indicating if there was a barcode match</li>" +
+            "<li>Matched barcode sequence (empty if read did not match one of the barcodes)</li>  " +
+            "</ul>" +
+            "If there is no match but we're close to the threshold of calling it a match, we output the barcode that would have been " +
+            "matched but in lower case.  Threshold values can be adjusted to accommodate barcode sequence mismatches from the reads." +
+            "The ExtractIlluminaBarcodes program also produces a metrics file and indicates the number of matches (and mismatches)" +
+            " between the barcode reads and the actual barcodes.  These metrics are provided both per-barcode and per lane and can be found in the BaseCalls directory.</p>" +
+            "<p>For poorly matching barcodes, the order of specification of barcodes can cause arbitrary output differences.</p>" +
+            "" +
+            "<h4>Usage example:</h4> " +
+                "<pre>" +
+                "java -jar picard.jar ExtractIlluminaBarcodes -h \\<br />" +
+        "              BASECALLS_DIR=s_1_1101.bcl \\<br />" +
```

Comment:
Same comment as above about .bcl extension in directory name. If that's how we expect them to be named that's fine but my understanding from your answer above is that it's not. If so these should all be corrected.